subcommand/sync-catalog/command_test.go

```diff
@@ -0,0 +1,210 @@
+	require.NoError(t, err)
+
+	exitChan := runCommandAsynchronously(&cmd, []string{
+		// change the write interval, so we can see changes in Consul quicker
+		"-consul-write-interval", "1s",
+	})
+
+	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	retry.RunWith(timer, t, func(r *retry.R) {
+		services, _, err := testAgent.Client().Catalog().Services(nil)
+		require.NoError(r, err)
+		require.Len(r, services, 2)
+		require.Contains(r, services, "foo")
+	})
+
+	// restart sync with -add-k8s-namespace-suffix
+	cmd.interrupt()
+	require.Equal(t, 0, <-exitChan)
+	exitChan = runCommandAsynchronously(&cmd, []string{
+		"-consul-write-interval", "1s",
+		"-add-k8s-namespace-suffix",
+	})
+
+	defer func() {
+		cmd.interrupt()
+		require.Equal(t, 0, <-exitChan)
+	}()
+
+	// check that the name of the service is now namespaced
+	retry.RunWith(timer, t, func(r *retry.R) {
+		services, _, err := testAgent.Client().Catalog().Services(nil)
+		require.NoError(r, err)
+		require.Len(r, services, 2)
+		require.Contains(r, services, "foo_default")
+	})
+}
+
+// Test that services with same name but in different namespaces
+// get registered as different services in consul
+// when using -add-k8s-namespace-suffix
+func TestCommand_Run_ToConsulTwoServicesSameNameDifferentNamespace(t *testing.T) {
+	t.Parallel()
+
+	k8s, testAgent := completeSetup(t)
+	defer testAgent.Shutdown()
+
+	// Run the command.
+	ui := cli.NewMockUi()
+	cmd := Command{
+		UI:           ui,
+		clientset:    k8s,
+		consulClient: testAgent.Client(),
+	}
+
+	// create two services in k8s
+	_, err := k8s.CoreV1().Services("bar").Create(testutil.LBService("foo", "1.1.1.1"))
+	require.NoError(t, err)
+
+	_, err = k8s.CoreV1().Services("baz").Create(testutil.LBService("foo", "2.2.2.2"))
+	require.NoError(t, err)
+
+	exitChan := runCommandAsynchronously(&cmd, []string{
+		"-consul-write-interval", "1s",
+		"-add-k8s-namespace-suffix",
+	})
+
+	defer func() {
+		cmd.interrupt()
+		require.Equal(t, 0, <-exitChan)
+	}()
+
+	// check that the name of the service is namespaced
+	timer := &retry.Timer{Timeout: 10 * time.Second, Wait: 500 * time.Millisecond}
+	retry.RunWith(timer, t, func(r *retry.R) {
+		svc, _, err := testAgent.Client().Catalog().Service("foo_bar", "", nil)
+		require.NoError(r, err)
+		require.Len(r, svc, 1)
+		require.Equal(r, "1.1.1.1", svc[0].ServiceAddress)
+		svc, _, err = testAgent.Client().Catalog().Service("foo_baz", "", nil)
+		require.NoError(r, err)
+		require.Len(r, svc, 1)
+		require.Equal(r, "2.2.2.2", svc[0].ServiceAddress)
+	})
+}
+
+// Set up test consul agent and fake kubernetes cluster client
+func completeSetup(t *testing.T) (*fake.Clientset, *agent.TestAgent) {
+	k8s := fake.NewSimpleClientset()
+	a := agent.NewTestAgent(t, t.Name(), `primary_datacenter = "dc1"`)
+
+	return k8s, a
+}
+
+// This function starts the command asynchronously and returns a blocking chan.
+// When finished, the command will send its exit code to the channel.
```

Comment:
What happens if the command doesn't terminate? Could that cause the tests that use this to run indefinitely?

Reply:
That's a great question. Since this function is for running sync-catalog command which is long-running, it's the responsibility of the caller to terminate it by calling `cmd.interrupt`. I've added a comment clarifying that.